pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java

```diff
@@ -114,6 +117,12 @@ public RealtimeSegmentImpl(Schema schema, int capacity, String tableName, String
     dictionaryMap = new HashMap<String, BaseOnHeapMutableDictionary>();
     maxNumberOfMultivaluesMap = new HashMap<String, Integer>();
     outgoingTimeColumnName = dataSchema.getTimeFieldSpec().getOutgoingTimeColumnName();
+
+    for (final String column : noDictionaryColumns) {
+      if (canBeNoDictionaryColumnInConsumingSegment(column, invertedIndexColumns)) {
```

Comment:
Not clear what you are asking for here, but I could not name the method any longer :-)

the method call determines if this column can be a no dictionary column for consuming segment. If yes, then we add it to the list, otherwise we end up creating dictionary for the column in the consuming segment. In the completed segment, this column will continue to have be a no-dictionary column. I will add comments saying so.